scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -67,4 +74,9 @@ def _decode(self, body, encoding):
                 body = zlib.decompress(body, -15)
         if encoding == b'br' and b'br' in ACCEPTED_ENCODINGS:
             body = brotli.decompress(body)
+        if encoding == b'zstd' and b'zstd' in ACCEPTED_ENCODINGS:
+            # Using its streaming API since its simple API could handle only cases
+            # where there is content size data embedded in the frame
+            reader = zstandard.ZstdDecompressor().stream_reader(io.BytesIO(body))
```

Comment:
hey! I wonder if it makes sense to use a context manager, to close the reader after it is used. This is very minor, as 
it is not doing much (https://github.com/indygreg/python-zstandard/blob/53b71dc3f96961564c9c140bf88b0aa118589247/zstandard/cffi.py#L1937), but it still may clean up some references earlier.